File: plugins/plugin-elizacloud/python/elizaos_plugin_elizacloud/models/embeddings.py
from elizaos_plugin_elizacloud.providers import ElizaCloudClient
from elizaos_plugin_elizacloud.types import ElizaCloudConfig, TextEmbeddingParams


async def handle_text_embedding(
    config: ElizaCloudConfig,
    text: str,
) -> list[float]:
    async with ElizaCloudClient(config) as client:
        params = TextEmbeddingParams(text=text)
        result = await client.generate_embedding(params)
        if isinstance(result[0], float):
            return result  # type: ignore[return-value]
        return result[0]


async def handle_batch_text_embedding(
    config: ElizaCloudConfig,
    texts: list[str],
) -> list[list[float]]:
    if not texts:
        return []

    async with ElizaCloudClient(config) as client:
        params = TextEmbeddingParams(texts=texts)
        result = await client.generate_embedding(params)
        if isinstance(result[0], list):
            return result  # type: ignore[return-value]
        return [result]  # type: ignore[list-item]

File: plugins/plugin-elizacloud/rust/src/models/embeddings.rs
#![allow(missing_docs)]

use crate::error::Result;
use crate::providers::ElizaCloudClient;
use crate::types::{ElizaCloudConfig, TextEmbeddingParams};

pub async fn handle_text_embedding(config: ElizaCloudConfig, text: String) -> Result<Vec<f32>> {
    let client = ElizaCloudClient::new(config)?;
    let params = TextEmbeddingParams::single(text);
    let embeddings = client.generate_embedding(params).await?;
    Ok(embeddings.into_iter().next().unwrap_or_default())
}

pub async fn handle_batch_text_embedding(
    config: ElizaCloudConfig,
    texts: Vec<String>,
) -> Result<Vec<Vec<f32>>> {
    if texts.is_empty() {
        return Ok(vec![]);
    }
    let client = ElizaCloudClient::new(config)?;
    let params = TextEmbeddingParams::batch(texts);
    client.generate_embedding(params).await
}
